src/asciidoctor-kroki.js

```diff
@@ -67,13 +71,24 @@ const processKroki = (processor, parent, attrs, diagramType, diagramText, contex
   delete blockAttrs.title
   delete blockAttrs.caption
   delete blockAttrs.opts
-  const inline = attrs['inline-option'] === ''
-  if (inline) {
-    blockAttrs['inline-option'] = ''
-  }
-  const interactive = attrs['interactive-option'] === ''
-  if (interactive) {
-    blockAttrs['interactive-option'] = ''
+
+  // `choices` is a list of pairs of input option value names and output option names
+  // First these are compared with the incoming block/macro attributes.
+  // If there is no match, they are compared with the kroki default settings in the document attributes.
+  var opt
+  choices.forEach(([name, option]) => {
+    if (attrs[name + '-option'] === '') {
+      opt = option
+    }
+  })
+  opt || choices.forEach(([name, option]) => {
+    if (parent.document.getAttribute('kroki-default-options') === name) {
+      opt = option
+    }
+  })
```

Comment:
I can read it :smile: 

I would suggest:

```js
const availableOptions = ['inline', 'interactive', 'none']
```

```suggestion

  let currentOption
  for (availableOption of availableOptions) {
    if (attrs[`${availableOption}-option`] === '') {
      currentOption = availableOption
    }
  }
  if (!currentOption) {
    for (availableOption of availableOptions) {
      if (parent.document.getAttribute('kroki-default-options') === availableOption) {
        currentOption = availableOption
      }
    }
  }
  if (currentOption && currentOption !== 'none') {
    blockAttrs[`${currentOption}-option`] = ''
  }
```

And probably extract the code in a function:

```js
const currentOption = getCurrentOption(attrs, parent.document)
if (currentOption && currentOption !== 'none') {
  blockAttrs[`${currentOption}-option`] = ''
}
```

Reply:
A separate function lets the code be a lot simpler!